Integrations/python/deephaven/learn/gather/__init__.py

```diff
@@ -78,45 +93,25 @@ def table_to_numpy_2d(row_set, col_set, order = Layout.ROW_MAJOR, dtype:np.dtype
     :return: A NumPy ndarray
     """
 
-    global Layout
-
-    if order in [Layout.ROW_MAJOR, Layout.C, True]:
-        flag = True
-    elif order in [Layout.COLUMN_MAJOR, Layout.FORTRAN, False]:
-        flag = False
-    else:
-        raise ValueError("Invalid major order.  Please use an True, False, or an enum value from MemoryLayout.")
+    if not(MemoryLayout.has_value(order)):
+        raise ValueError("Invalid major order {order}.  Please use an enum value from MemoryLayout.")
 
-    if dtype == bool or dtype == float or dtype == int:
-        dtype = convert_to_numpy_dtype(dtype)
+    dtype = convert_to_numpy_dtype(dtype)
 
     if dtype == np.byte:
-
-        buffer = _gatherer.tensorBuffer2DByte(row_set, col_set, flag)
-
+        buffer = _gatherer.tensorBuffer2DByte(row_set, col_set, order)
```

Comment:
I think this might not be doing what you want.  I'm guessing python is doing truthiness on `order`, and since order is always non-null, I think it is defaulting to `True`.  I think you want:

```suggestion
        buffer = _gatherer.tensorBuffer2DByte(row_set, col_set, order.is_row_major)
```

I could be wrong on this one.  If I'm right, apply the change everywhere.

https://www.freecodecamp.org/news/truthy-and-falsy-values-in-python/